Base/Base.Static/FileLogger.cpp

```diff
@@ -0,0 +1,210 @@
+#include "FileLogger.hpp"
+
+using namespace Logger;
+
+std::string FileLogger::stringifyLogLvl(const LogLevel level)
+{
+    std::string result =        "NONE";
+    const static std::map<LogLevel, std::string> LevelMap = {
+        {LogLevel::ERROR,       "ERROR"},
+        {LogLevel::WARNING,     "WARNING"},
+        {LogLevel::INFO,        "INFO"},
+        {LogLevel::DEBUG,       "DEBUG"},
+    };
+
+    auto it = LevelMap.find(level);
+    if (it != LevelMap.end())
+    {
+        result = it->second;
+    }
+    return result;
+}
+
+FileLogger::FileLogger()
+{
+    _Thread = std::thread(&FileLogger::run, this);
+}
+
+FileLogger::~FileLogger()
```

Comment:
So looking at the loop at https://github.com/f1nal3/Juniorgram/pull/335/files#diff-22f48a4279d67dad9eae041bff9b63b01d60098017b7908a3fb4be2b5be1680cR127-R137
When will this thread join?